docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,422 @@
+- *requestHandler*: <string> one of the available request handlers or <function> for custom one
+- *responseHandler*: <string> one of the available response handlers or <function> for custom one
+- *editor*: <string> one of the available editors or Editor class for custom one
+- *editorConfig*: object holding editor parameters
+- *options.showTimePicker*: <bool> show or hide time picker (defaults to true)
+- *options.showQueryBar*: <bool> show or hide query bar (defaults to true)
+- *options.showFilterBar*: <bool> show or hide filter bar (defaults to true)
+- *options.showIndexSelection*: <bool> show or hide index selection (defaults to true)
+- *isExperimental*: <bool> mark visualization as experimental (defaults to false)
+
+
+Each of the factories have some of the custom parameters, which will be described below.
+
+[[development-base-visualization-type]]
+==== Base Visualization Type
+The base visualization type does not make any assumptions about the rendering technology you are going to use and
+works with pure Javascript. It's the visualization type we recommend to use.
+
+You need to provide an object with constructor, render function which will be called every time options 
+or data change and destroy function which will be called to cleanup.
+ 
+The render function receives the data object and status object which tells what actually changed.
+Render function needs to return a promise, which should be resolved once the visualization is done rendering.
+
+Status object has the following properties: `vis`, `aggs`, `resize`, `data`. Each of them is set to true if the matching
+object changed since last call to the render function or set to false otherwise. You can use it to make your 
+visualization rendering more efficient.
+
+
+image::images/visualize-flow.png[Main Flow]
+
+- Your visualizations constructor will get called with `vis` object and element to which it should render.
+At this point you should prepare everything for rendering, but not render yet
+- <visualize> component monitors `appState`, `uiState` and `vis` for changes
+- on change <visualize> will call your `requestHandler`, which should return new data (or in case nothing changed
+the old data). Implementing request handler is optional, you might use on of the provided.
+- response from `requestHandler` will get passed to `responseHandler`. IT should convert raw data to something that
+can be consumed by visualization. Implementing responseHandler is optional, you might use of of the provided.
+- `ResizeMonitor` monitors the <visualize> for size change
+- on new data from `responseHandler` or on size change your visualization render method gets called.
+It should return a promise and resolve once its done rendering.
+- <visualize> will emit `renderComplete` event on the element once your render methods promise is resolved.
+- visualization should call `updateState()` any time something changed that requires to re-render or fetch new data.
```

Comment:
]> The visualization should call `vis.updateState()` any time something has changed that requires to re-render or fetch new data.